gateway-service/src/main/java/org/zowe/apiml/gateway/zaas/ZaasController.java

```diff
@@ -116,4 +123,48 @@ public ResponseEntity<Object> getZoweJwt(@RequestAttribute(AUTH_SOURCE_ATTR) Aut
         }
     }
 
+    @PostMapping(path = "safIdt", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Provides SAF Identity Token for authenticated user.")
+    public ResponseEntity<Object> getSafIdToken(@RequestBody TicketRequest ticketRequest, @RequestAttribute(AUTH_SOURCE_PARSED_ATTR) AuthSource.Parsed authSourceParsed) {
+
```

Comment:
This looks good to me. It is just too long for the controller. I would move it to the new or existing service.